File: src/main/java/com/telmomenezes/synthetic/cli/Compare.java
package com.telmomenezes.synthetic.cli;

import com.telmomenezes.synthetic.MetricsBag;
import com.telmomenezes.synthetic.Net;


public class Compare extends Command {
	
	@Override
	public String name() {return "compare";}
	
	
	@Override
	public String help() {
		String help = "Compares two networks.\n";
		help += "$ synt compare -inet <network1> -inet2 <network2>\n";
		help += "Optional parameters:\n";
		help += "-undir if network is undirected.\n";
		help += "-bins <n> distribution bins (default is 100).\n";
		return help;
    }
	
	
	@Override
    public boolean run() throws SynCliException {
        String netfile1 = getStringParam("inet");
        String netfile2 = getStringParam("inet2");
        int bins = getIntegerParam("bins", 100);
        boolean directed = !paramExists("undir");
        boolean par = paramExists("par");
        
        Net net1 = Net.load(netfile1, directed, par);
        Net net2 = Net.load(netfile2, directed, par);
        
     	System.out.println("NET: " + netfile1);
     	System.out.println(net1);
        MetricsBag bag1 = new MetricsBag(net1, bins);
        System.out.println(bag1);
        
        System.out.println("\n\n");
        
        System.out.println("NET: " + netfile2);
        System.out.println(net2);
        MetricsBag bag2 = new MetricsBag(net2, null, null, bins, bag1);
        System.out.println(bag2);
        
        return true;
    }
}

File: netgens/commands/compare.py
from netgens.net import load_net
from netgens.stats import StatsSet
from netgens.commands.command import *


class Compare(Command):
    def __init__(self):
        Command.__init__(self)
        self.name = 'compare'
        self.mandatory_args = ['inet', 'inet2']

    def help(self):
        lines = ['Compares two networks.',
                 '$ netgens compare -inet <network1> -inet2 <network2>',
                 'Optional parameters:',
                 '-undir if network is undirected',
                 '-bins <n> distribution bins (default is 100)']
        return '\n'.join(lines)

    def run(self, args):
        self.error_msg = None

        netfile1 = args['inet']
        netfile2 = args['inet2']

        bins = arg_with_default(args, 'bins', 100)
        directed = not args['undir']

        # load nets
        net1 = load_net(netfile1, directed)
        net2 = load_net(netfile2, directed)

        stat_dist_types = get_stat_dist_types(args)

        print('first network: %s' % netfile1)
        print(net1)
        stats1 = StatsSet(net1, stat_dist_types, bins, ref_stats=None)
        print(stats1)

        print('\n\n')

        print('second network: %s' % netfile2)
        print(net2)
        stats2 = StatsSet(net1, stat_dist_types, bins, ref_stats=None)
        print(stats2)

        return True
